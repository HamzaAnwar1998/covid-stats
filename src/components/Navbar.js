import React from 'react';
import { Icon } from 'react-icons-kit';
import { mic } from 'react-icons-kit/feather/mic';
import { settings } from 'react-icons-kit/feather/settings';

const Navbar = () => (
  <nav className="navbox">
    <h2 className="logo">COVID-STATS</h2>
    <ul className="list">
      <li className="item">
        <Icon icon={mic} size={18} />
      </li>
      <li className="item">
        <Icon icon={settings} size={18} />
      </li>
    </ul>
  </nav>
);

export default Navbar;
