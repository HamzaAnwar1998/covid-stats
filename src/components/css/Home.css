.card{
    width: 200px;
    height: auto;
    display: flex;
    flex-direction: column;
    align-items: center;
    border: 1px solid #e4e4e4;
    margin: 10px 0;
    border-radius: 18px;
    padding: 10px;
    background-color: #453095 !important;
    color: #fff;
    overflow: hidden;
}

.card .flag-img{
    width: 150px;
    height: 100px;
    margin: 0 auto;
}

.card .flag-img img{
    width: 100%;
    height: 100%;
}

.country{
    font-size: 20px;
    margin-top: 12px;
    text-align: center;
}

.active{
    font-size: 14px;
    font-weight: 600;
}

.details-link{
    text-decoration: none;
    font-size: 12px;
    color: #fff;
}