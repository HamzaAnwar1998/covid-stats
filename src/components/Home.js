/* eslint-disable max-len */
/* eslint-disable import/no-named-as-default */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import './css/Home.css';
import { Icon } from 'react-icons-kit';
import { arrowRightCircle } from 'react-icons-kit/feather/arrowRightCircle';
import { Link, useLocation } from 'react-router-dom';
import getCovidDataAction from '../redux/actions';

const Home = () => {
  const location = useLocation();
  const query = new URLSearchParams(location.search);
  const search = query.get('search') || '';

  const data = useSelector((state) => state.covidReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    if (data.length === 0) {
      dispatch(getCovidDataAction());
    }
  }, []);
  const filter = data.filter((individualData) => individualData.Country.includes(search.toLowerCase()));
  return filter.map((individualData) => (
    <div key={individualData.ID} className="card">
      <div className="flag-img">
        <img src={individualData.Flag} alt="flag" />
      </div>
      <h2 className="country">{individualData.Country}</h2>
      <p className="active">
        Active Cases:
        {' '}
        {individualData.ActiveCases}
      </p>
      <Link className="details-link" to={{ pathname: `/country/${individualData.Country}` }}>
        <span>
          More details
          {' '}
          <Icon icon={arrowRightCircle} size={20} />
        </span>
      </Link>
    </div>
  ));
};

export default Home;
