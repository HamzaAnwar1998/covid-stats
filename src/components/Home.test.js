import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Navbar from './Navbar';
import Home from './Home';

describe('Render components', () => {
  test('render NavBar', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Navbar /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render Home', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Home /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
