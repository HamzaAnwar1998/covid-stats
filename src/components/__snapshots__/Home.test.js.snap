// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render Home 1`] = `null`;

exports[`Render components render NavBar 1`] = `
<nav
  className="navbox"
>
  <h2>
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      COVID-STATS
    </a>
  </h2>
  <ul
    className="list"
  >
    <li
      className="item"
    >
      <i
        fill="currentColor"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <svg
          fill="none"
          height={18}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 24 24"
          width={18}
        >
          <path
            d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
          />
          <path
            d="M19 10v2a7 7 0 0 1-14 0v-2"
          />
          <line
            x1="12"
            x2="12"
            y1="19"
            y2="23"
          />
          <line
            x1="8"
            x2="16"
            y1="23"
            y2="23"
          />
        </svg>
      </i>
    </li>
    <li
      className="item"
    >
      <i
        fill="currentColor"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <svg
          fill="none"
          height={18}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 24 24"
          width={18}
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </i>
    </li>
  </ul>
</nav>
`;
